arch/arm/mach-imx/imx_secondary_boot.c

```diff
@@ -25,10 +25,15 @@ static int do_imx_secondary_boot(cmd_tbl_t *cmdtp, int flag,
 
 	/* If we just want to retrieve current value */
 	if (argc == 1) {
-		arm_smccc_smc(IMX_SIP_SRC, IMX_SIP_SRC_IS_SECONDARY_BOOT, 0, 0,
-			      0, 0, 0, 0, &res);
+		if (CONFIG_IS_ENABLED(ARCH_IMX8M)) {
+			arm_smccc_smc(IMX_SIP_SRC,
+				      IMX_SIP_SRC_IS_SECONDARY_BOOT, 0, 0,
+				      0, 0, 0, 0, &res);
```

Comment:
On the imx6 case res will be defined but not used.

Reply:
Yes, will fix